server/game/cards/02_SHD/upgrades/MandalorianArmor.ts

```diff
@@ -0,0 +1,27 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { Card } from '../../../core/card/Card';
+import { UpgradeCard } from '../../../core/card/UpgradeCard';
+import { Trait } from '../../../core/Constants';
+
+export default class MandalorianArmor extends UpgradeCard {
+    protected override getImplementationId () {
+        return {
+            id: '3514010297',
+            internalName: 'mandalorian-armor',
+        };
+    }
+
+    public override setupCardAbilities () {
+        this.setAttachCondition((card: Card) => !card.hasSomeTrait(Trait.Vehicle));
+
+        this.addWhenPlayedAbility({
+            title: 'Shield attached unit when Mandalorian',
+            targetResolver: {
```

Comment:
This is incorrect, as currently written it will allow the player to give a shield to _any_ Mandalorian unit on the field.

`targetResolver` should only be used when the player has a choice of targets to select from. In this case the target is known automatically (the attached card), so we can use `immediateEffect` instead and explicitly specify the target. We do that inside of a conditional to check if it's Mandalorian:

```typescript
immediateEffect: AbilityHelper.immediateEffects.conditional({
    condition: (context) => context.source.parentCard?.hasSomeTrait(Trait.Mandalorian),
    onTrue: AbilityHelper.immediateEffects.giveShield((context) => ({ target: context.source.parentCard })),
    onFalse: AbilityHelper.immediateEffects.noAction()
});